include/crocoddyl/multibody/costs/frame-placement.hpp

```diff
@@ -164,32 +161,10 @@ struct CostDataFramePlacementTpl : public CostDataAbstractTpl<_Scalar> {
 
   template <template <typename Scalar> class Model>
   CostDataFramePlacementTpl(Model<Scalar>* const model, DataCollectorAbstract* const data)
-      : Base(model, data),
-        J(6, model->get_state()->get_nv()),
-        rJf(6, 6),
-        fJf(6, model->get_state()->get_nv()),
-        Arr_J(6, model->get_state()->get_nv()) {
-    r.setZero();
-    J.setZero();
-    rJf.setZero();
-    fJf.setZero();
+      : Base(model, data), Arr_J(6, model->get_state()->get_nv()) {
     Arr_J.setZero();
-    // Check that proper shared data has been passed
-    DataCollectorMultibodyTpl<Scalar>* d = dynamic_cast<DataCollectorMultibodyTpl<Scalar>*>(shared);
-    if (d == NULL) {
-      throw_pretty("Invalid argument: the shared data should be derived from DataCollectorMultibody");
-    }
-
-    // Avoids data casting at runtime
-    pinocchio = d->pinocchio;
   }
 
-  pinocchio::DataTpl<Scalar>* pinocchio;
-  Vector6s r;
-  pinocchio::SE3Tpl<Scalar> rMf;
-  Matrix6xs J;
-  Matrix6s rJf;
-  Matrix6xs fJf;
```

Comment:
The "change to reference" strategy should apply to these quantities as well. For example, CosttDataFramePlacement->rMf is now broken.

Reply:
I would ask you to wait for the next PR, that will deprecate this classes. This is the most pragmatical approach, however, if you believe this is needed, then I would ask you if you can handle it. Unfortunately, I cannot afford to handle this alone.